### DISCUSSION TITLE
Support for 304 Not Modified

### DISCUSSION THREAD

**Discussion by ranger-ross** [TanStack/query#4454] [Q&A] [answered]
Hello, I have a query in my application that is fetching a large json object from an api (that I do not own).

I have a the `staleTime` set fairly low because I want this data to be updated frequently.
Everything is working fine, but looking in the DevTools Network tab I'm using a lot of bandwidth for the revalidation.

The API I am using supports [If-Modified-Since](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since) header, but I am not exactly sure to go able implementing it with React Query.

I see 2 problems:
1. How do I get the previous fetch timestamp and how do I get that to my fetch function? 🤔 
2. When the api responds with a 304 status, it will not return a response body, so what should my fetch function return?

I guess I could implement this using external state, but if React Query has a first class way to do this that would be great.

**Comment by TkDodo** [comment, marked as answer]
This is not something react-query can or will support out of the box because it is not something that react-query is concerned with. React-Query is an async state manager - it is not tied to data fetching, and thus cannot know about http headers.

Isn't that something that browser would do automatically?

For example:
- if the server responds with an [ETag](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag), browsers will send the ETag back as an `if-none-match` header.
- if the server responds with a [Last-Modified](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified) header, browsers will send that timestamp back as `If-Modified-since`

at least that's what I though so far. Would be weird if clients had to manage those dates by themselves 🤔

**Reply by ranger-ross** [reply]
I see, I think you are correct. (although the Chrome DevTools is a bit misleading, since it reports the original request's status code)

I think my mistake was including `'cache-control': 'no-cache'` in my request headers by mistake.
Thanks for the quick reply 💯
